grails-app/services/org/pih/warehouse/requisition/RequisitionService.groovy

```diff
@@ -504,6 +506,13 @@ class RequisitionService {
         return requisition
     }
 
+    void deleteRequisition(StockMovement stockMovement, Requisition requisition) {
+        OutboundStockMovement associatedStockMovement = OutboundStockMovement.get(stockMovement.id)
```

Comment:
since you fetch it once in the `StockMovementService`, can't you just pass it instead of re-fetching it here?